cardinal/ecs/world.go

```diff
@@ -102,38 +103,41 @@ func (w *World) AddSystems(s ...System) {
 	}
 }
 
-// RegisterComponents attempts to initialize the given slice of components with a WorldAccessor.
-// This will give components the ability to access their own data.
-func (w *World) RegisterComponents(components ...component.IComponentType) error {
-	if w.stateIsLoaded {
+func RegisterComponent[T component.IAbstractComponent](world *World) error {
+	if world.stateIsLoaded {
 		panic("cannot register components after loading game state")
 	}
-	if w.isComponentsRegistered {
-		return ErrorComponentRegistrationMustHappenOnce
-	}
-	w.isComponentsRegistered = true
-	w.registeredComponents = append(w.registeredComponents, SignerComp)
-	w.registeredComponents = append(w.registeredComponents, components...)
 
-	for i, c := range w.registeredComponents {
-		id := component.TypeID(i + 1)
-		if err := c.SetID(id); err != nil {
+	//Register SignerComponent component if it does not exist
+	_, err := world.GetComponentByName(SignerComp.Name())
```

Comment:
can we just do this in `NewWorld` constructor where we are registering the persona systems and such?

Reply:
This was done because we use a Global `SignerComponent` that is owned by the global namespace. 

I made it private. Should look better now.